packages/webos/LS2Request/LS2Request.js

```diff
@@ -53,15 +65,27 @@ export default class LS2Request {
 		// eslint-disable-next-line no-undef
 		this.bridge = new PalmServiceBridge();
 		this.bridge.onservicecallback = this.callback.bind(this, onSuccess, onFailure, onComplete);
+		if (timeout) {
+			// timer will be cleared when the callback is run, so this should only fire if the timeout threshold is reached
+			this.timer = new Job(() => {
```

Comment:
Let's declare `Job` outside so that we don't initiate every time `send()` is invoked. Something like this

```
timeOutJob = new Job( ... )
...
send () {
  ... 
  if (timeout) {
    timeOutJob.startAfter(timeout);
  }
  ...
}
```

Reply:
Can there only be one outstanding LS2 request at a time?

Reply:
Or did you mean at the class level?

Reply:
at class level

Reply:
The issue I see is that `onTimeout` is passed to `send()` so at the class level, I can't make the job function use it.

Reply:
You could pass arguments for the callback function as a second parameter in `startAfter`. So something like this: 

```
timeOutJob = new Job(({onTimeout, timeout}) => {
...
this.timeOutJob.startAfter(timeout, {onTimeout, timeout});
```

Reply:
:genius: